File: python3/1004.py
from typing import List

class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l, r = 0, 0
        ans, cnt = 0, 0
        n = len(nums)
        while r < n:
            if nums[r] == 0:
                cnt += 1
            while cnt > k:
                if nums[l] == 0:
                    cnt -= 1
                l += 1
            ans = max(ans, r - l + 1)
            r += 1
        return ans

File: cpp/1004.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l = 0, r = 0;
        int ans = 0, cnt = 0;
        int len = nums.size();
        while (r < len) {
            if (!nums[r]) {
                ++cnt;
            }
            while (cnt > k) {
                if (!nums[l]) {
                    --cnt;
                }
                ++l;
            }
            ans = max(ans, r - l + 1);
            ++r;
        }
        return ans;
    }
};